core/chains/evm/gas/rollups/l1_oracle.go

```diff
@@ -0,0 +1,313 @@
+
+	l1GasPriceAddress   string
+	gasPriceMethod      string
+	l1GasPriceMethodAbi abi.ABI
+	l1GasPriceMu        sync.RWMutex
+	l1GasPrice          *assets.Wei
+
+	l1GasCostAddress   string
+	gasCostMethod      string
+	l1GasCostMethodAbi abi.ABI
+
+	chInitialised chan struct{}
+	chStop        services.StopChan
+	chDone        chan struct{}
+}
+
+const (
+	// ArbGasInfoAddress is the address of the "Precompiled contract that exists in every Arbitrum chain."
+	// https://github.com/OffchainLabs/nitro/blob/f7645453cfc77bf3e3644ea1ac031eff629df325/contracts/src/precompiles/ArbGasInfo.sol
+	ArbGasInfoAddress = "0x000000000000000000000000000000000000006C"
+	// ArbGasInfo_getL1BaseFeeEstimate is the a hex encoded call to:
+	// `function getL1BaseFeeEstimate() external view returns (uint256);`
+	ArbGasInfo_getL1BaseFeeEstimate = "getL1BaseFeeEstimate"
+	// NodeInterfaceAddress is the address of the precompiled contract that is only available through RPC
+	// https://github.com/OffchainLabs/nitro/blob/e815395d2e91fb17f4634cad72198f6de79c6e61/nodeInterface/NodeInterface.go#L37
+	ArbNodeInterfaceAddress = "0x00000000000000000000000000000000000000C8"
+	// ArbGasInfo_getPricesInArbGas is the a hex encoded call to:
+	// `function gasEstimateL1Component(address to, bool contractCreation, bytes calldata data) external payable returns (uint64 gasEstimateForL1, uint256 baseFee, uint256 l1BaseFeeEstimate);`
+	ArbNodeInterface_gasEstimateL1Component = "gasEstimateL1Component"
+
+	// OPGasOracleAddress is the address of the precompiled contract that exists on OP stack chain.
+	// This is the case for Optimism and Base.
+	OPGasOracleAddress = "0x420000000000000000000000000000000000000F"
+	// OPGasOracle_l1BaseFee is a hex encoded call to:
+	// `function l1BaseFee() external view returns (uint256);`
+	OPGasOracle_l1BaseFee = "l1BaseFee"
+	// OPGasOracle_getL1Fee is a hex encoded call to:
+	// `function getL1Fee(bytes) external view returns (uint256);`
+	OPGasOracle_getL1Fee = "getL1Fee"
+
+	// ScrollGasOracleAddress is the address of the precompiled contract that exists on Scroll chain.
+	ScrollGasOracleAddress = "0x5300000000000000000000000000000000000002"
+	// ScrollGasOracle_l1BaseFee is a hex encoded call to:
+	// `function l1BaseFee() external view returns (uint256);`
+	ScrollGasOracle_l1BaseFee = "l1BaseFee"
+	// ScrollGasOracle_getL1Fee is a hex encoded call to:
+	// `function getL1Fee(bytes) external view returns (uint256);`
+	ScrollGasOracle_getL1Fee = "getL1Fee"
+
+	// GasOracleAddress is the address of the precompiled contract that exists on Kroma chain.
+	// This is the case for Kroma.
+	KromaGasOracleAddress = "0x4200000000000000000000000000000000000005"
+	// GasOracle_l1BaseFee is the a hex encoded call to:
+	// `function l1BaseFee() external view returns (uint256);`
+	KromaGasOracle_l1BaseFee = "l1BaseFee"
+
+	// Interval at which to poll for L1BaseFee. A good starting point is the L1 block time.
+	PollPeriod = 12 * time.Second
```

Comment:
I know this is same as previously.
But 12 seconds could sometimes mean our price estimates can be 11-12 seconds stale.
Could we make this a configurable setting, or if not, atleast change the default to 6 seconds?

Reply:
Looks like we don't have an existing config for gas estimator poll periods so I just updated this to 6. It's hard coded in other places too like [here](https://github.com/smartcontractkit/chainlink/blob/59d1c9954662200f787968ce7d89c53036e33dd0/core/chains/evm/gas/arbitrum_estimator.go#L61) and [here](https://github.com/smartcontractkit/chainlink/blob/59d1c9954662200f787968ce7d89c53036e33dd0/core/chains/evm/gas/suggested_price_estimator.go#L52). Maybe I can handle adding a configurable `PollPeriod` under `[GasEstimator]` configs in another ticket for all estimators.